{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table style=\"border: 2px solid white;\">\n",
       "<tr>\n",
       "<td style=\"vertical-align: top; border: 0px solid white\">\n",
       "<h3 style=\"text-align: left;\">Client</h3>\n",
       "<ul style=\"text-align: left; list-style: none; margin: 0; padding: 0;\">\n",
       "  <li><b>Scheduler: </b>inproc://10.0.0.32/45504/18</li>\n",
       "  <li><b>Dashboard: </b><a href='http://10.0.0.32/45504/18:8787/status' target='_blank'>http://10.0.0.32/45504/18:8787/status</a>\n",
       "</ul>\n",
       "</td>\n",
       "<td style=\"vertical-align: top; border: 0px solid white\">\n",
       "<h3 style=\"text-align: left;\">Cluster</h3>\n",
       "<ul style=\"text-align: left; list-style:none; margin: 0; padding: 0;\">\n",
       "  <li><b>Workers: </b>4</li>\n",
       "  <li><b>Cores: </b>12</li>\n",
       "  <li><b>Memory: </b>16.00 GB</li>\n",
       "</ul>\n",
       "</td>\n",
       "</tr>\n",
       "</table>"
      ],
      "text/plain": [
       "<Client: 'inproc://10.0.0.32/45504/18' processes=4 threads=12, memory=16.00 GB>"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%matplotlib inline\n",
    "import numpy as np\n",
    "from scipy import stats\n",
    "from blimpy import Waterfall\n",
    "from blimpy.utils import rebin\n",
    "from matplotlib import pyplot as plt\n",
    "from bisect import bisect_left\n",
    "from tqdm import tqdm\n",
    "import dask.array as da\n",
    "import h5py\n",
    "from time import time\n",
    "\n",
    "from utils import *\n",
    "\n",
    "fil_path = \"data/filterbanks/\"\n",
    "h5_path = \"data/h5/\"\n",
    "\n",
    "test_fil = fil_path + \"blc20_guppi_57991_48899_3C161_0007.gpuspec.0000.fil\"\n",
    "\n",
    "fri_obs = h5_path + \"GBT_57532_09539_HIP56445_fine.h5\"\n",
    "\n",
    "plt_args = {\n",
    "            'aspect':'auto',\n",
    "            'origin':'lower',\n",
    "            'rasterized':True,\n",
    "            'interpolation':'nearest',\n",
    "            'cmap':'viridis'\n",
    "            }\n",
    "try:\n",
    "    client.close()\n",
    "except NameError:\n",
    "    pass\n",
    "from dask.distributed import Client, progress\n",
    "client = Client(processes=False, threads_per_worker=3,\n",
    "                n_workers=4, memory_limit='4GB')\n",
    "client"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "blimpy.file_wrapper INFO     Skipping loading data ...\n",
      "\n",
      "--- File Info ---\n",
      "b'DIMENSION_LABELS' : [b'frequency' b'feed_id' b'time']\n",
      "     b'az_start' :                              0.0\n",
      "    b'data_type' :                                1\n",
      "         b'fch1' :                    1926.26953125\n",
      "         b'foff' :           -2.835503418452676e-06\n",
      "   b'machine_id' :                               20\n",
      "        b'nbits' :                               32\n",
      "       b'nchans' :                        318230528\n",
      "         b'nifs' :                                1\n",
      "  b'source_name' :                      b'HIP56445'\n",
      "      b'src_dej' :                     3d03m34.006s\n",
      "      b'src_raj' :                    11h34m21.699s\n",
      " b'telescope_id' :                                6\n",
      "        b'tsamp' :                     17.986224128\n",
      "       b'tstart' :                57532.11040509259\n",
      "     b'za_start' :                              0.0\n",
      "\n",
      "Num ints in file :                               16\n",
      "      File shape :               (16, 1, 318230528)\n",
      "--- Selection Info ---\n",
      "Data selection shape :               (16, 1, 318230528)\n",
      "Minimum freq (MHz) :               1023.9257812499999\n",
      "Maximum freq (MHz) :                    1926.26953125\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/blimpy/file_wrapper.py:326: H5pyDeprecationWarning: The default file mode will change to 'r' (read-only) in h5py 3.0. To suppress this warning, pass the mode you need to h5py.File(), or set the global default h5.get_config().default_file_mode, or set the environment variable H5PY_DEFAULT_READONLY=1. Available modes are: 'r', 'r+', 'w', 'w-'/'x', 'a'. See the docs for details.\n",
      "  self.h5 = h5py.File(self.filename)\n"
     ]
    }
   ],
   "source": [
    "# Show Info\n",
    "wf = Waterfall(fri_obs, load_data=False)\n",
    "wf.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "wf = Waterfall(fri_obs, f_stop=1080, max_load=2)\n",
    "wf.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1033216"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "channel_len = np.int(np.round(187.5/64/abs(wf.header[b'foff'])))\n",
    "channel_len"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.figure()\n",
    "wf.plot_waterfall(f_start=0, f_stop=1080)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "freqs, data = wf.grab_data(f_start=0, f_stop=1080)\n",
    "print(data.shape)\n",
    "freqs = freqs[::-1]\n",
    "data = data[:,::-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(freqs)/channel_len"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "data = data[:, :16 * 1033216]\n",
    "freqs = freqs[:16 * 1033216]\n",
    "\n",
    "integrated = np.sum(data, axis=0)/len(data)\n",
    "for n in np.nonzero(integrated > 10**13):\n",
    "    integrated[n] = (integrated[n-1] + integrated[n+1]) /2\n",
    "plt.figure()\n",
    "plt.plot(freqs, np.log(integrated))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "c = 6\n",
    "\n",
    "window = data[:, channel_len*(c):channel_len*(c+1)]\n",
    "plot_segment(np.log(window))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "window"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "res = []\n",
    "for chan in tqdm(range(15)):\n",
    "    res.append([])\n",
    "    window = data[:, channel_len*(chan):channel_len*(chan+1)]\n",
    "    window_f = freqs[channel_len*(chan):channel_len*(chan+1)]\n",
    "    for i in range(0, (len(window[0])//200*200), 100):\n",
    "        test_data = window[:, i:i+200]\n",
    "        s, p = norm_test(test_data)\n",
    "        if p < 1e-25:\n",
    "            res[chan].append((window_f[i], s, p))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from multiprocessing import Pool\n",
    "from time import time\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "\n",
    "def threshold_hits(chan):\n",
    "    res = list()\n",
    "    window = data[:, channel_len*(chan):channel_len*(chan+1)]\n",
    "    window_f = freqs[channel_len*(chan):channel_len*(chan+1)]\n",
    "    for i in range(0, (len(window[0])//200*200), 100):\n",
    "        test_data = window[:, i:i+200]\n",
    "        s, p = norm_test(test_data)\n",
    "        if p < 1e-25:\n",
    "            res.append((window_f[i], s, p))\n",
    "    return res\n",
    "\n",
    "start = time()\n",
    "with Pool(12) as p:\n",
    "    chan_hits = p.map(threshold_hits, range(16))\n",
    "end = time()\n",
    "print(end-start)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "show_stamp_f(freqs, data, 1025.0100805927195)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "hits = [len(e) for e in chan_hits]\n",
    "print(hits)\n",
    "print(sum(hits))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sorted_hits = sorted(res[7], key=lambda x: x[2])\n",
    "sorted_hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "top = [x[0] for x in sorted_hits[:20]]\n",
    "top"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "for i in top:\n",
    "    print(i)\n",
    "    show_stamp(window, i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "res[:15]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.figure()\n",
    "plt.imshow(data[:, 8:8+128])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"pfb512coef.txt\", \"r\") as f:\n",
    "    coef_file = f.read()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "lines = coef_file.splitlines()\n",
    "filter_coefs = []\n",
    "for line in lines:\n",
    "    filter_coefs.append(float(line))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "coefs = np.array(filter_coefs)/2**17"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.figure()\n",
    "plt.plot(coefs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from numpy import fft\n",
    "l = 2**16\n",
    "f = fft.fft(coefs, l)\n",
    "plt.figure()\n",
    "plt.plot(np.log(np.abs(f)**2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "stacked = np.reshape(integrated, (15, 1033216))\n",
    "plt.figure()\n",
    "for i in range(15):\n",
    "    plt.plot(np.log(stacked[i]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "rebined = rebin(stacked, 1, 64)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "plt.figure()\n",
    "plt.imshow(np.log(rebined), **plt_args)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "model_shape = np.sum(stacked, axis=0)/15\n",
    "model_shape[model_shape > 10**10] = np.mean(model_shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "plt.figure()\n",
    "plt.plot(model_shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<tr>\n",
       "<td>\n",
       "<table>\n",
       "  <thead>\n",
       "    <tr><td> </td><th> Array </th><th> Chunk </th></tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr><th> Bytes </th><td> 20.37 GB </td> <td> 115.72 MB </td></tr>\n",
       "    <tr><th> Shape </th><td> (16, 1, 318230528) </td> <td> (2, 1, 14465024) </td></tr>\n",
       "    <tr><th> Count </th><td> 177 Tasks </td><td> 176 Chunks </td></tr>\n",
       "    <tr><th> Type </th><td> float32 </td><td> numpy.ndarray </td></tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</td>\n",
       "<td>\n",
       "<svg width=\"194\" height=\"90\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"10\" y1=\"0\" x2=\"24\" y2=\"14\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"10\" y1=\"25\" x2=\"24\" y2=\"40\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"10\" y1=\"0\" x2=\"10\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"11\" y1=\"1\" x2=\"11\" y2=\"27\" />\n",
       "  <line x1=\"13\" y1=\"3\" x2=\"13\" y2=\"29\" />\n",
       "  <line x1=\"15\" y1=\"5\" x2=\"15\" y2=\"31\" />\n",
       "  <line x1=\"17\" y1=\"7\" x2=\"17\" y2=\"32\" />\n",
       "  <line x1=\"19\" y1=\"9\" x2=\"19\" y2=\"34\" />\n",
       "  <line x1=\"21\" y1=\"11\" x2=\"21\" y2=\"36\" />\n",
       "  <line x1=\"23\" y1=\"13\" x2=\"23\" y2=\"38\" />\n",
       "  <line x1=\"24\" y1=\"14\" x2=\"24\" y2=\"40\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"10.000000,0.000000 24.948598,14.948598 24.948598,40.361214 10.000000,25.412617\" style=\"fill:#ECB172A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"10\" y1=\"0\" x2=\"130\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"11\" y1=\"1\" x2=\"131\" y2=\"1\" />\n",
       "  <line x1=\"13\" y1=\"3\" x2=\"133\" y2=\"3\" />\n",
       "  <line x1=\"15\" y1=\"5\" x2=\"135\" y2=\"5\" />\n",
       "  <line x1=\"17\" y1=\"7\" x2=\"137\" y2=\"7\" />\n",
       "  <line x1=\"19\" y1=\"9\" x2=\"139\" y2=\"9\" />\n",
       "  <line x1=\"21\" y1=\"11\" x2=\"141\" y2=\"11\" />\n",
       "  <line x1=\"23\" y1=\"13\" x2=\"143\" y2=\"13\" />\n",
       "  <line x1=\"24\" y1=\"14\" x2=\"144\" y2=\"14\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"10\" y1=\"0\" x2=\"24\" y2=\"14\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"15\" y1=\"0\" x2=\"30\" y2=\"14\" />\n",
       "  <line x1=\"20\" y1=\"0\" x2=\"35\" y2=\"14\" />\n",
       "  <line x1=\"26\" y1=\"0\" x2=\"41\" y2=\"14\" />\n",
       "  <line x1=\"31\" y1=\"0\" x2=\"46\" y2=\"14\" />\n",
       "  <line x1=\"37\" y1=\"0\" x2=\"52\" y2=\"14\" />\n",
       "  <line x1=\"42\" y1=\"0\" x2=\"57\" y2=\"14\" />\n",
       "  <line x1=\"48\" y1=\"0\" x2=\"63\" y2=\"14\" />\n",
       "  <line x1=\"53\" y1=\"0\" x2=\"68\" y2=\"14\" />\n",
       "  <line x1=\"59\" y1=\"0\" x2=\"74\" y2=\"14\" />\n",
       "  <line x1=\"64\" y1=\"0\" x2=\"79\" y2=\"14\" />\n",
       "  <line x1=\"70\" y1=\"0\" x2=\"84\" y2=\"14\" />\n",
       "  <line x1=\"75\" y1=\"0\" x2=\"90\" y2=\"14\" />\n",
       "  <line x1=\"80\" y1=\"0\" x2=\"95\" y2=\"14\" />\n",
       "  <line x1=\"86\" y1=\"0\" x2=\"101\" y2=\"14\" />\n",
       "  <line x1=\"91\" y1=\"0\" x2=\"106\" y2=\"14\" />\n",
       "  <line x1=\"97\" y1=\"0\" x2=\"112\" y2=\"14\" />\n",
       "  <line x1=\"102\" y1=\"0\" x2=\"117\" y2=\"14\" />\n",
       "  <line x1=\"108\" y1=\"0\" x2=\"123\" y2=\"14\" />\n",
       "  <line x1=\"113\" y1=\"0\" x2=\"128\" y2=\"14\" />\n",
       "  <line x1=\"119\" y1=\"0\" x2=\"134\" y2=\"14\" />\n",
       "  <line x1=\"124\" y1=\"0\" x2=\"139\" y2=\"14\" />\n",
       "  <line x1=\"130\" y1=\"0\" x2=\"144\" y2=\"14\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"10.000000,0.000000 130.000000,0.000000 144.948598,14.948598 24.948598,14.948598\" style=\"fill:#ECB172A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"24\" y1=\"14\" x2=\"144\" y2=\"14\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"24\" y1=\"40\" x2=\"144\" y2=\"40\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"24\" y1=\"14\" x2=\"24\" y2=\"40\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"30\" y1=\"14\" x2=\"30\" y2=\"40\" />\n",
       "  <line x1=\"35\" y1=\"14\" x2=\"35\" y2=\"40\" />\n",
       "  <line x1=\"41\" y1=\"14\" x2=\"41\" y2=\"40\" />\n",
       "  <line x1=\"46\" y1=\"14\" x2=\"46\" y2=\"40\" />\n",
       "  <line x1=\"52\" y1=\"14\" x2=\"52\" y2=\"40\" />\n",
       "  <line x1=\"57\" y1=\"14\" x2=\"57\" y2=\"40\" />\n",
       "  <line x1=\"63\" y1=\"14\" x2=\"63\" y2=\"40\" />\n",
       "  <line x1=\"68\" y1=\"14\" x2=\"68\" y2=\"40\" />\n",
       "  <line x1=\"74\" y1=\"14\" x2=\"74\" y2=\"40\" />\n",
       "  <line x1=\"79\" y1=\"14\" x2=\"79\" y2=\"40\" />\n",
       "  <line x1=\"84\" y1=\"14\" x2=\"84\" y2=\"40\" />\n",
       "  <line x1=\"90\" y1=\"14\" x2=\"90\" y2=\"40\" />\n",
       "  <line x1=\"95\" y1=\"14\" x2=\"95\" y2=\"40\" />\n",
       "  <line x1=\"101\" y1=\"14\" x2=\"101\" y2=\"40\" />\n",
       "  <line x1=\"106\" y1=\"14\" x2=\"106\" y2=\"40\" />\n",
       "  <line x1=\"112\" y1=\"14\" x2=\"112\" y2=\"40\" />\n",
       "  <line x1=\"117\" y1=\"14\" x2=\"117\" y2=\"40\" />\n",
       "  <line x1=\"123\" y1=\"14\" x2=\"123\" y2=\"40\" />\n",
       "  <line x1=\"128\" y1=\"14\" x2=\"128\" y2=\"40\" />\n",
       "  <line x1=\"134\" y1=\"14\" x2=\"134\" y2=\"40\" />\n",
       "  <line x1=\"139\" y1=\"14\" x2=\"139\" y2=\"40\" />\n",
       "  <line x1=\"144\" y1=\"14\" x2=\"144\" y2=\"40\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"24.948598,14.948598 144.948598,14.948598 144.948598,40.361214 24.948598,40.361214\" style=\"fill:#ECB172A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"84.948598\" y=\"60.361214\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >318230528</text>\n",
       "  <text x=\"164.948598\" y=\"27.654906\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(0,164.948598,27.654906)\">1</text>\n",
       "  <text x=\"7.474299\" y=\"52.886915\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(45,7.474299,52.886915)\">16</text>\n",
       "</svg>\n",
       "</td>\n",
       "</tr>\n",
       "</table>"
      ],
      "text/plain": [
       "dask.array<array, shape=(16, 1, 318230528), dtype=float32, chunksize=(2, 1, 14465024), chunktype=numpy.ndarray>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h5_file = h5py.File(fri_obs, \"r\")\n",
    "a = da.from_array(h5_file[\"data\"], chunks=(2, 1, channel_len*14))\n",
    "a = a\n",
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/37 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/38 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/39 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/40 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/41 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/42 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/43 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/44 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/47 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/45 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/46 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/49 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/48 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/50 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/51 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/21\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/22\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/22' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/22' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/22' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/21\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/23\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: connect() didn't finish in time\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/52 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/55 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/53 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/59 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/54 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/57 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/56 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/58 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/60 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/23\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/22\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/22' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/22' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/22' after 10 s: connect() didn't finish in time\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/61 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/64 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/62 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/63 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/65 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/66 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/67 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/68 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/69 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/70 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/71 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/72 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/73 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/74 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/75 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/76 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/77 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/78 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/79 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/80 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/81 remote=inproc://10.0.0.32/45504/18>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/82 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/83 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/84 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/85 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/86 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/87 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/88 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/89 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/90 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/94 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/95 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/96 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/97 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/98 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/99 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/91 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/92 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/93 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/23\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/21\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/22\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/22' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/22' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/22' after 10 s: connect() didn't finish in time\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/100 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/101 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/104 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/109 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/102 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/107 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/103 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/108 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/105 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/106 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/110 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/111 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/112 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/113 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/114 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/115 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/116 remote=inproc://10.0.0.32/45504/18>\n",
      "tornado.application - ERROR - Exception in callback functools.partial(<function wrap.<locals>.null_wrapper at 0x2125dbf28>, <Task finished coro=<Worker.heartbeat() done, defined at /Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py:855> exception=OSError(\"Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\",)>)\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/ioloop.py\", line 758, in _run_callback\n",
      "    ret = callback()\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/stack_context.py\", line 300, in null_wrapper\n",
      "    return fn(*args, **kwargs)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/ioloop.py\", line 779, in _discard_future_result\n",
      "    future.result()\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 864, in heartbeat\n",
      "    metrics=await self.get_metrics(),\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/117 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/120 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/118 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/119 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/121 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/122 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/123 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/124 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/125 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/126 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/127 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/128 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/129 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/130 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/131 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/135 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/132 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/136 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/133 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/138 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/139 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/140 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/141 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/142 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/143 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/144 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/145 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/146 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/147 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/134 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/137 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/22\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/22' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/22' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/22' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/23\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: connect() didn't finish in time\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/21\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/22\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/22' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/22' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/22' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/148 remote=inproc://10.0.0.32/45504/23>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/152 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/149 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/153 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/150 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/155 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/151 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/154 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/156 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/157 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/158 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/159 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/160 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/161 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/162 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/163 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/164 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/165 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/166 remote=inproc://10.0.0.32/45504/23>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/170 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/167 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/168 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/169 remote=inproc://10.0.0.32/45504/18>\n",
      "tornado.application - ERROR - Exception in callback functools.partial(<function wrap.<locals>.null_wrapper at 0x2507516a8>, <Task finished coro=<Worker.heartbeat() done, defined at /Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py:855> exception=OSError(\"Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\",)>)\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/ioloop.py\", line 758, in _run_callback\n",
      "    ret = callback()\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/stack_context.py\", line 300, in null_wrapper\n",
      "    return fn(*args, **kwargs)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/ioloop.py\", line 779, in _discard_future_result\n",
      "    future.result()\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 864, in heartbeat\n",
      "    metrics=await self.get_metrics(),\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "tornado.application - ERROR - Exception in callback functools.partial(<function wrap.<locals>.null_wrapper at 0x2bee916a8>, <Task finished coro=<Worker.heartbeat() done, defined at /Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py:855> exception=OSError(\"Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\",)>)\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/ioloop.py\", line 758, in _run_callback\n",
      "    ret = callback()\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/stack_context.py\", line 300, in null_wrapper\n",
      "    return fn(*args, **kwargs)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/ioloop.py\", line 779, in _discard_future_result\n",
      "    future.result()\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 864, in heartbeat\n",
      "    metrics=await self.get_metrics(),\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/23\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/21\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: connect() didn't finish in time\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/171 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/172 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/173 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/174 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/175 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/176 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/177 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/178 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/179 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/180 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/181 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/182 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/183 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/23\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/23' after 10 s: connect() didn't finish in time\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/184 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/187 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/185 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/186 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/188 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/189 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/190 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/191 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/192 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/193 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/194 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/195 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/196 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/197 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/198 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/199 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/200 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/202 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/201 remote=inproc://10.0.0.32/45504/23>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/203 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/204 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/205 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/206 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/207 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/208 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/209 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/210 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/211 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/212 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/213 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/214 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/215 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/216 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/217 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/218 remote=inproc://10.0.0.32/45504/18>\n",
      "tornado.application - ERROR - Exception in callback functools.partial(<function wrap.<locals>.null_wrapper at 0x12ddac268>, <Task finished coro=<Worker.memory_monitor() done, defined at /Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py:2504> exception=TypeError('Could not serialize object of type Dataset.', '<HDF5 dataset \"data\": shape (16, 1, 318230528), type \"<f4\">')>)\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/ioloop.py\", line 758, in _run_callback\n",
      "    ret = callback()\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/stack_context.py\", line 300, in null_wrapper\n",
      "    return fn(*args, **kwargs)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/ioloop.py\", line 779, in _discard_future_result\n",
      "    future.result()\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2566, in memory_monitor\n",
      "    k, v, weight = self.data.fast.evict()\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/zict/lru.py\", line 90, in evict\n",
      "    cb(k, v)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/zict/buffer.py\", line 52, in fast_to_slow\n",
      "    self.slow[key] = value\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/zict/func.py\", line 42, in __setitem__\n",
      "    self.d[key] = self.dump(value)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/protocol/serialize.py\", line 441, in serialize_bytelist\n",
      "    header, frames = serialize(x, **kwargs)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/protocol/serialize.py\", line 210, in serialize\n",
      "    raise TypeError(msg, str(x)[:10000])\n",
      "TypeError: ('Could not serialize object of type Dataset.', '<HDF5 dataset \"data\": shape (16, 1, 318230528), type \"<f4\">')\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/219 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/220 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/221 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/222 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/223 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/224 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/225 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/226 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/227 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/228 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/229 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/230 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/231 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/232 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/233 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/234 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/235 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/236 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/238 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/237 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/239 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/240 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/241 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/244 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/245 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/242 remote=inproc://10.0.0.32/45504/22>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/247 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/243 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/246 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/250 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/248 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/249 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/251 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/252 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/253 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/254 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/255 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/256 remote=inproc://10.0.0.32/45504/18>\n",
      "tornado.application - ERROR - Exception in callback functools.partial(<function wrap.<locals>.null_wrapper at 0x25073b9d8>, <Task finished coro=<Worker.heartbeat() done, defined at /Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py:855> exception=OSError(\"Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\",)>)\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/ioloop.py\", line 758, in _run_callback\n",
      "    ret = callback()\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/stack_context.py\", line 300, in null_wrapper\n",
      "    return fn(*args, **kwargs)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/ioloop.py\", line 779, in _discard_future_result\n",
      "    future.result()\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 864, in heartbeat\n",
      "    metrics=await self.get_metrics(),\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/257 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/258 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/260 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/261 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/262 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/263 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/259 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/264 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/265 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/266 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/267 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/268 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/272 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/269 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/270 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/271 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/273 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/274 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/275 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/276 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/277 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/278 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.worker - ERROR - Handle missing dep failed, retrying\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2019, in handle_missing_dep\n",
      "    who_has = await self.scheduler.who_has(keys=list(deps))\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 735, in send_recv_from_rpc\n",
      "    comm = await self.pool.connect(self.addr)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/18' after 10 s: connect() didn't finish in time\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/287 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/288 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/289 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/293 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/290 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/291 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/292 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/294 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/295 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/296 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/297 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/298 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/299 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/300 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/301 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/302 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/303 remote=inproc://10.0.0.32/45504/18>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/304 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/308 remote=inproc://10.0.0.32/45504/23>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/305 remote=inproc://10.0.0.32/45504/22>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/306 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/307 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 13% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 13% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 12% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/322 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/323 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/324 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 11% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/325 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/21\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: connect() didn't finish in time\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/326 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/327 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/328 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/329 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/330 remote=inproc://10.0.0.32/45504/21>\n",
      "tornado.application - ERROR - Exception in callback functools.partial(<function wrap.<locals>.null_wrapper at 0x12da616a8>, <Task finished coro=<Worker.memory_monitor() done, defined at /Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py:2504> exception=TypeError('Could not serialize object of type Dataset.', '<HDF5 dataset \"data\": shape (16, 1, 318230528), type \"<f4\">')>)\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/ioloop.py\", line 758, in _run_callback\n",
      "    ret = callback()\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/stack_context.py\", line 300, in null_wrapper\n",
      "    return fn(*args, **kwargs)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/tornado/ioloop.py\", line 779, in _discard_future_result\n",
      "    future.result()\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 2566, in memory_monitor\n",
      "    k, v, weight = self.data.fast.evict()\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/zict/lru.py\", line 90, in evict\n",
      "    cb(k, v)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/zict/buffer.py\", line 52, in fast_to_slow\n",
      "    self.slow[key] = value\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/zict/func.py\", line 42, in __setitem__\n",
      "    self.d[key] = self.dump(value)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/protocol/serialize.py\", line 441, in serialize_bytelist\n",
      "    header, frames = serialize(x, **kwargs)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/protocol/serialize.py\", line 210, in serialize\n",
      "    raise TypeError(msg, str(x)[:10000])\n",
      "TypeError: ('Could not serialize object of type Dataset.', '<HDF5 dataset \"data\": shape (16, 1, 318230528), type \"<f4\">')\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/331 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.utils_perf - WARNING - full garbage collections took 10% CPU time recently (threshold: 10%)\n",
      "distributed.worker - ERROR - Worker stream died during communication: inproc://10.0.0.32/45504/21\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 221, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: connect() didn't finish in time\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 1881, in gather_dep\n",
      "    self.rpc, deps, worker, who=self.address\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/worker.py\", line 3099, in get_data_from_worker\n",
      "    comm = await rpc.connect(worker)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/core.py\", line 869, in connect\n",
      "    connection_args=self.connection_args,\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 230, in connect\n",
      "    _raise(error)\n",
      "  File \"/Users/yuhongc/PycharmProjects/BSRC_Env/lib/python3.6/site-packages/distributed/comm/core.py\", line 204, in _raise\n",
      "    raise IOError(msg)\n",
      "OSError: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: Timed out trying to connect to 'inproc://10.0.0.32/45504/21' after 10 s: connect() didn't finish in time\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/332 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.comm.inproc - WARNING - Closing dangling queue in <InProc  local=inproc://10.0.0.32/45504/333 remote=inproc://10.0.0.32/45504/21>\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.07 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Worker is at 131% memory usage. Pausing worker.  Process memory: 5.28 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.28 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.48 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 6.10 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Worker is at 162% memory usage. Pausing worker.  Process memory: 6.48 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 6.57 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 6.58 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 6.59 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.99 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.99 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.99 GB -- Worker memory limit: 4.00 GB\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.99 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.99 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.99 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.99 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.99 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.99 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.99 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 6.01 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 6.01 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 6.25 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 6.25 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 6.69 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.88 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.92 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.92 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.53 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.53 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.53 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.53 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.53 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.53 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.53 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.53 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 3.86 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 3.86 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 3.86 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 3.86 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 3.86 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 3.86 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 3.86 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 3.86 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Worker is at 50% memory usage. Resuming worker. Process memory: 2.01 GB -- Worker memory limit: 4.00 GB\n",
      "distributed.worker - WARNING - Worker is at 50% memory usage. Resuming worker. Process memory: 2.01 GB -- Worker memory limit: 4.00 GB\n"
     ]
    }
   ],
   "source": [
    "da.to_npy_stack(\"original_a\", a, axis=2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "start = time()\n",
    "means = da.mean(a, axis=2)\n",
    "# means_data = means.compute()\n",
    "# means_data\n",
    "end = time()\n",
    "\n",
    "print(f\"{end-start}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "means = da.from_array(np.reshape(means_data, (16,1,1)))\n",
    "means"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "308/14"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "normalized_a = da.divide(a.T, means.T).T\n",
    "normalized_a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "da.to_npy_stack('normalized_a/', normalized_a, axis=2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(normalized_a.shape)\n",
    "print(normalized_a.chunks)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "start = time()\n",
    "normalized_a_data = normalized_a.compute()\n",
    "end = time()\n",
    "print(f\"{end-start}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "start = time()\n",
    "\n",
    "original_a_data = a[:, 0, :16 * 1033216].compute()\n",
    "\n",
    "end = time()\n",
    "print(f\"{end-start}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "read_a = da.from_array(original_a_data)\n",
    "read_and_normalize = da.divide(read_a, means_data)\n",
    "\n",
    "start = time()\n",
    "normalized_a_data = read_and_normalize.compute()\n",
    "end = time()\n",
    "print(f\"{end-start}\")\n",
    "\n",
    "# 1.6420118808746338"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "del normalized_a_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "start = time()\n",
    "\n",
    "numpy_normalized_a_data = original_a_data / means_data\n",
    "\n",
    "end = time()\n",
    "print(f\"{end-start}\")\n",
    "\n",
    "# 0.6659681797027588"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "del numpy_normalized_a_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "numpy_normalized_a_data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plot_segment(normalized_a_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plot_segment(original_a_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(normalized_a_data.shape == numpy_normalized_a_data.shape)\n",
    "print(np.allclose(normalized_a_data, numpy_normalized_a_data))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "BSRC_Env",
   "language": "python",
   "name": "bsrc_env"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
